src/NServiceBus.Core/Routing/AutomaticSubscriptions/AutoSubscribe.cs

```diff
@@ -30,16 +30,12 @@ protected internal override void Setup(FeatureConfigurationContext context)
             }
 
             var conventions = context.Settings.Get<Conventions>();
-            var transportInfrastructure = context.Settings.Get<TransportInfrastructure>();
-            var requireExplicitRouting = transportInfrastructure.OutboundRoutingPolicy.Publishes == OutboundRoutingType.Unicast;
-            var publishers = context.Routing.Publishers;
 
             context.RegisterStartupTask(b =>
             {
                 var handlerRegistry = b.Build<MessageHandlerRegistry>();
                 var messageTypesHandled = GetMessageTypesHandledByThisEndpoint(handlerRegistry, conventions, settings);
-                var typesToSubscribe = messageTypesHandled.Where(eventType => !requireExplicitRouting || publishers.GetPublisherFor(eventType).Any()).ToList();
```

Comment:
Why is okay to not filter `messageTypesHandled` now? I didn't see anything your changes that addressed this. Was this actually uneeded?

Reply:
@bording the filter removed events where no publisher was registered when using message driven pubsub. This was done, as `Subscribe` throws an exception if you try to subscribe to a type without a matching publisher route. As we want to log those cases, we're now just catching the exception instead of preventing the call in the first place.